clang-tools-extra/test/clang-tidy/checkers/readability/use-builtin-literals-cpp23.cpp

```diff
@@ -0,0 +1,11 @@
+// RUN: %check_clang_tidy -std=c++23-or-later %s readability-use-builtin-literals %t
+
+#include <stddef.h>
```

Comment:
Including system headers in test code is generally frowned upon, because the test ecosystem cannot control what the current Clang will pick up from where, and these can be platform-specific. Just for `size_t`, the "official" definition can be easily reproduced. (For more complex cases, there are special headers in the test ecosystem that deal precisely with simulating the contents of system headers.)

```cpp
namespace std {
  using size_t = decltype(sizeof(void*));
}
```